honeybee_energy/lib/materials.py

```diff
@@ -0,0 +1,138 @@
+"""Establish the defauly materials within the honeybee_energy library."""
+from honeybee_energy.material.opaque import EnergyMaterial
+from honeybee_energy.material.glazing import EnergyWindowMaterialGlazing
+from honeybee_energy.material.gas import EnergyWindowMaterialGas
+
+from ._loadconstructions import _idf_opaque_materials, _idf_window_materials
+
+
+# auto-generate materials and constructions if they were not loaded
+# generic opaque materials
+if 'Generic Brick' not in _idf_opaque_materials:
```

Comment:
My suggestion is to put all the default materials that will be used in this method in a single dictionary and then call them from the dictionary in the checks. That will help to being able to see all the default materials in the same place and will editing or reusing them easier in the future.

Reply:
I don't follow exactly what you mean here since this code is not a method and the default materials are already in a dictionary (the `_idf_opaque_materials`).  However, I have realized that this is likely a place where EAFP will be better performance than LBYL and so I am going to rewrite this code now to reflect that.  Let me know if that is what you meant.

Reply:
That's not what I meant. I mean something like this for all your placeholder materials that you are using in this file.

https://github.com/mostaphaRoudsari/honeybee-radiance/blob/command/honeybee_radiance/modifierset.py#L15

Reply:
@mostaphaRoudsari ,
I think that you might not have understood what the `_idf_opaque_materials` dictionary is since it is virtually equivalent to what you point to in that file.  The only difference is that the dictionary is loaded from a .idf file, which users can edit if they ever want to change the honeybee default materials.  The materials.py file where you made this comment is just adding missing keys to this dictionary in the event that someone accidentally deletes them entirely from the .idf file.

Reply:
Thanks! I didn't mean it for `_idf_opaque_materials` dictionary but for all the other ones that you generate inside this file using `EnergyMaterial()`. I can send a PR later that shows what I meant. For now it works great and my suggestion was only meant to help the maintenance in the long run. We will see if it will be an issue at all!